components/patient-data/impl/src/main/java/org/phenotips/data/internal/controller/MetaDataController.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.phenotips.data.internal.controller;
+
+import org.phenotips.data.DictionaryPatientData;
+import org.phenotips.data.Patient;
+import org.phenotips.data.PatientData;
+import org.phenotips.data.PatientDataController;
+
+import org.xwiki.component.annotation.Component;
+
+import java.text.SimpleDateFormat;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+
+import com.xpn.xwiki.doc.XWikiDocument;
+
+import net.sf.json.JSONObject;
+
+/**
+ * Exposes the metainfo about the document.
+ *
+ * @version $Id$
+ * @since 1.0M10
+ */
+@Component(roles = { PatientDataController.class })
+@Named("metadata")
+@Singleton
+public class MetaDataController extends AbstractSimpleController implements PatientDataController<String>
+{
+    private static final String UNKNOWN_USER = "Unknown user";
+
+    private static final String DOCUMENT_NAME = "doc.name";
+
+    private static final String DOCUMENT_NAME_STRING = "report_id";
+
+    private static final String REFERRER = "referrer";
+
+    private static final String CREATION_DATE = "creationDate";
+
+    private static final String AUTHOR = "author";
+
+    private static final String AUTHOR_STRING = "last_modified_by";
+
+    private static final String DATE = "date";
+
+    private static final String DATE_STRING = "last_modification_date";
+
+    private static final String CONTROLLER_NAME = "metadata";
+
+    @Inject
+    private Logger logger;
+
+    @Override
+    public PatientData<String> load(Patient patient)
+    {
+        try {
+            XWikiDocument doc = (XWikiDocument) this.documentAccessBridge.getDocument(patient.getDocument());
+
+            Map<String, String> result = new LinkedHashMap<String, String>();
+
+            SimpleDateFormat format = new SimpleDateFormat("yyyy.MM.dd");
```

Comment:
I took this custom date format from DataToCellConverter.java, where it is used for the same fields. Should we also change format there to standard ISO date-time just to be consistent?